### ISSUE TITLE
luacheck requires a real path to a directory?

### ISSUE THREAD

**Issue by Totktonada** [mpeterv/luacheck#208]
I found that under the following circumstances I need to pass a real directory path (with resolved symlinks) to the luacheck executable in order to don't miss warnings:

1. `.luacheckrc` in a project root have `include_files` with some globs.
2. The current directory is non-real path to the project root.
3. The argument to luacheck is the same non-real path.

How to reproduce:

```sh
$ mkdir -p "${HOME}"/tmp/test-luacheck/aaa  # project root
$ cd "${HOME}"/tmp/test-luacheck/aaa
$ echo 'foo = 42' > foo.lua
$ echo "include_files = {'**/*.lua'}" > .luacheckrc
$ luacheck --config .luacheckrc "${HOME}"/tmp/test-luacheck/aaa
Checking /home/alex/tmp/test-luacheck/aaa/foo.lua 1 warning

    /home/alex/tmp/test-luacheck/aaa/foo.lua:1:1: setting non-standard global variable foo

Total: 1 warning / 0 errors in 1 file
$ cd ..
$ ln -s aaa bbb
$ cd bbb
$ luacheck --config .luacheckrc "${HOME}"/tmp/test-luacheck/bbb
Total: 0 warnings / 0 errors in 0 files
$ cd .. && tree
.
├── aaa
│   └── foo.lua
└── bbb -> aaa

2 directories, 1 file
```

I step into this problem, because I use symlinks to ease access to directories that I use often. I'm not sure this is broad approach, but I hope it is valid. At least, it seems, it does not contradict with the current luacheck documentation.

(I'm aware about the [status](https://github.com/mpeterv/luacheck/issues/198) of the repository, but I think that issues should be filed here until the repository will be properly migrated: otherwise we'll have several issue numerations.)

**Comment by Totktonada** [comment]
I also found that even if I pass the real path, but the current working directory is different, the files are missed:

```sh
$ pwd
/home/alex/tmp/test-luacheck
$ mkdir ccc
$ cd ccc
$ luacheck --config ../aaa/.luacheckrc /home/alex/tmp/test-luacheck/aaa
Total: 0 warnings / 0 errors in 0 files
```

**Comment by ligurio** [comment]
I suggest to resolve specified path using `readlink`, it works:

```
Total: 1 warning / 0 errors in 1 file
sergeyb@pony:~/sources/MRG/ttt$ luacheck --config .luacheckrc /home/sergeyb/sources/MRG/ttt
Total: 0 warnings / 0 errors in 0 files
sergeyb@pony:~/sources/MRG/ttt$ luacheck --config .luacheckrc $(readlink /home/sergeyb/sources/MRG/ttt)
Checking /home/sergeyb/sources/MRG/tt/test.lua    1 warning

    /home/sergeyb/sources/MRG/tt/test.lua:1:1: setting non-standard global variable a

Total: 1 warning / 0 errors in 1 file
sergeyb@pony:~/sources/MRG/ttt$ 
```

**Comment by Totktonada** [comment]
@ligurio It seems it works only when .luacheckrc can be found by upward directories traversal. See https://github.com/mpeterv/luacheck/issues/208#issuecomment-725147237.
